src/dotnet/MLSearch.Service/Bot/Tools/BotAuthSchemeHandler.cs

```diff
@@ -0,0 +1,79 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+public class BotAuthSchemeHandler : AuthenticationHandler<BotAuthenticationSchemeOptions>
+{
+    // This is a workaround implementation.
+    private readonly SigningCredentials StubSigningCredentials;
+    
+    public BotAuthSchemeHandler(
+        IOptionsMonitor<BotAuthenticationSchemeOptions> options,
+        ILoggerFactory logger,
+        UrlEncoder encoder) : base(options, logger, encoder)
+    {
+        StubSigningCredentials = options.CurrentValue.SigningCredentials;
+    }
+
+    protected async override Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        var principal = GetValidatedClaims(this.Context.Request);
+        if (principal == null) {
+            return AuthenticateResult.Fail("Authentication failed");
+        }
+        var ticket = new AuthenticationTicket(principal, this.Scheme.Name);
+        return AuthenticateResult.Success(ticket);
+    }
+    public ClaimsPrincipal GetValidatedClaims(HttpRequest request) {
+        const string BEARER_PREFIX = "Bearer ";
+        // TODO: Read the token from request headers/cookies
+        // Check that it's a valid session, depending on your implementation
+        string bearerToken = request.Headers.Authorization.Where(e=> e.StartsWith(BEARER_PREFIX)).FirstOrDefault();
+        if (bearerToken == null) {
+            return null;
+        }
+        var claims = VerifyToken(bearerToken.Substring(BEARER_PREFIX.Length));
```

Comment:
Here must be a warning about proper string comparison. I think StringComparison.Ordinal is what being expected here (and in the StartsWith call above).

Reply:
will fix.